src/main/java/org/minions/devfund/damianvp/battleship/ShipCreator.java

```diff
@@ -0,0 +1,30 @@
+package org.minions.devfund.damianvp.battleship;
+
+/**
+ * ShipCreator class.
+ */
+public class ShipCreator {
+    /**
+     * This method get the instance of specific class of ship.
+     * @param shipType string type of ship.
+     * @return specific ship instance.
+     */
+    public Ship getShip(final String shipType) {
+        switch (shipType) {
+            case "BattleShip":
+                return new BattleShip();
+            case "BattleCruiser":
+                return new BattleCruiser();
+            case "Cruiser":
+                return new Cruiser();
+            case "LightCruiser":
+                return new LightCruiser();
+            case "Destroyer":
+                return new Destroyer();
+            case "Submarine":
+                return new Submarine();
+            default:
+                return new EmptySea();
+        }
```

Comment:
Instead of using a switch 
You can use the following code.
```java
    interface ShipBuilder {

        /**
         * Builds a new ship.
         *
         * @return {@link Ship}.
         */
        Ship buildShip();
    }

    private static final Map<String, ShipBuilder> SHIP_FACTORY = new HashMap<>();
    static {
        SHIP_FACTORY.put("BattleShip", BattleShip::new);
        SHIP_FACTORY.put("BattleCruiser", BattleCruiser::new);
        SHIP_FACTORY.put("Cruiser", Cruiser::new);
        SHIP_FACTORY.put("LightCruiser", LightCruiser::new);
        SHIP_FACTORY.put("Destroyer", Destroyer::new);
        SHIP_FACTORY.put("Submarine", Submarine::new);
    }

    /**
     * This method get the instance of specific class of ship.
     * @param shipType string type of ship.
     * @return specific ship instance.
     */
    public Ship getShip(final String shipType) {
        return SHIP_FACTORY.getOrDefault(shipType, EmptySea::new).buildShip();
    }
```

Reply:
@MinionsGroup/minions-h 
You can take as example this implementation.

Reply:
cool, thanks for the example, I will apply